service/ioc-start.sh

```diff
@@ -15,5 +15,14 @@ done
 case "$(decode-reg build_info -q --slot ${devslot})" in
     afc-tim-receive*|afcv4_fofb_ctrl*|pbpm-gw*|bpm-gw*)
         systemctl restart afc-ioc@${devslot}.service
+        ;;&
+    bpm-gw*)
+        VSLOT1=$(/opt/afc-epics-ioc/iocBoot/iocutca/getSlot.sh $devslot)
+        VSLOT2=$(( VSLOT1 + 1 ))
+
+        services="rffe-ioc@${VSLOT1}.service"
+        [ "$VSLOT2" -lt 24 ] && services+=" rffe-ioc@${VSLOT2}.service"
+
+        systemctl restart $services
```

Comment:
I would prefer to keep this script working with `/bin/sh`.

Have a `launch_afc_ioc` variable, that's set for the first case (without `bpm-gw` now), and for the second case as well, after the rffe-ioc services have been started. Then do a conditional to decide on it.

Reply:
If you do this, add `|| true` to the `restart` command, since rffe-ioc might not be installed.

Reply:
> I would prefer to keep this script working with /bin/sh.

Just for the sake of learning, why do you have this preference? Depending on BASH for such a low level script can bring us problems?

> Have a `launch_afc_ioc` variable [...] Then do a conditional to decide on it.

I'm thinking it might be cleaner to make the conditional for RFFE instead. I'll give it a try to see if my intuition is right.

Reply:
Never mind... It starts to become unreadable because the lack of string comparison features:

```sh
start_rffe_ioc=false

synthesis="$(decode-reg build_info -q --slot ${devslot})"

case "$synthesis" in
    afc-tim-receive*|afcv4_fofb_ctrl*|pbpm-gw*|bpm-gw*)
        systemctl restart afc-ioc@${devslot}.service

        [ "${synthesis#bpm-gw}" != "${synthesis}" ] && start_rffe_ioc=true
        ;;
esac

if $start_rffe_ioc; then
    VSLOT1=$(/opt/afc-epics-ioc/iocBoot/iocutca/getSlot.sh $devslot)
    VSLOT2=$(( VSLOT1 + 1 ))

    systemctl start rffe-ioc@${VSLOT1}.service rffe-ioc@${VSLOT2}.service || true
fi

```